### ISSUE TITLE
Gewählter Zeitraum wird bei Switch in anderen Tab vergessen

### ISSUE THREAD

**Issue by oioix** [AndiLeni/statistics#88]
Cooles Addon
Schön wäre es, wenn der individuell gewählte Zeitraum bei Switch in einen anderen Reiter (z.B. von Statistik zu Seitenaufrufe) erhalten bliebe. 
LG, Chris

(enhancement)

**Comment by AndiLeni** [comment]
Die Tabs für die Navigation zwischen den Seiten wird von Redaxo selbst generiert. 
Ich bin mir nicht sicher, ob ich da einen Parameter für den Filter mitgeben kann.

Ich lasse das Issue offen und bin natürlich auch über Pull requests glücklich :-)

**Comment by oioix** [comment]
Hi Andreas, verstehe ;-)

Da gibt es sicher einen ExtensionPoint, wo man das eingestellte Datum (aus der Url extrahiert z.B. "&date_start=2022-02-15&date_end=2022-04-02") anhängen könnte.

Fragt sich nur welcher das ist.
Evtl. geht es auch mit JS.

**Comment by iceman-fx** [comment]
Mache es dir einfach und speichere die Auswahl in der session.
Dann kannst die dies über alle Seiten hinweg nutzen.

**Comment by AndiLeni** [comment]
Das ist eine sehr gute Idee :)

**Comment by oioix** [comment]
Ansonsten... das JS macht den Job, solange man nur zwischen den drei relevanten Tabs wechselt:

`// --------------------------------------------------------------------------------------------------------------------------
// -- START DOCUMENT READY --------------------------------------------------------------------------------------------------
$(document).ready(function() {
	
	
	// -------------------------------------------------------------------------------------
	// rex addon statistics -> attach filtered date to addon tab links
	// -------------------------------------------------------------------------------------
	var header = $('.rex-page-nav').find('ul').find('a');
	
	var linkStatsElem	= '';
	var linkStatsVal	= '';
	
	var linkPagesElem	= '';
	var linkPagesVal	= '';
	
	var linkRefererElem	= '';
	var linkRefererVal	= '';
	
	header.each(function(){
		var thisLink = $(this).attr('href');
		
		var inStrStats	= thisLink.includes('statistics/stats');
		var inStrPages	= thisLink.includes('statistics/pages');
		var inStrReferer = thisLink.includes('statistics/referer');
		
		if(inStrStats) {
			linkStatsElem = $(this);
			linkStatsVal = thisLink;
		}
		if(inStrPages) {
			linkPagesElem = $(this);
			linkPagesVal = thisLink;
		}
		if(inStrReferer) {
			linkRefererElem = $(this);
			linkRefererVal = thisLink;
		}
	});
	
	var filterStartDateElem = $('#statistics_datefilter_start');
	var filterEndDateElem	= $('#statistics_datefilter_end');
	
	// get current values
	function getStatFilterDates(){
		var filterStartDate = filterStartDateElem.val();
		var filterEndDate	= filterEndDateElem.val();
	
		if(filterStartDate && filterEndDate) {
			
			var attachDateVal = '&date_start='+filterStartDate+'&date_end='+filterEndDate;
			
			linkStatsElem.attr('href',linkStatsVal+attachDateVal);
			linkPagesElem.attr('href',linkStatsVal+attachDateVal);
			linkRefererElem.attr('href',linkStatsVal+attachDateVal);
		}
	}
	
	// run on pageload
	getStatFilterDates();
	
	// run on change
	filterStartDateElem.change(function(){
		getStatFilterDates();
	});
	filterEndDateElem.change(function(){
		getStatFilterDates();
	});
});`

**Comment by oioix** [comment]
Es liest die date-inputs aus. Wenn du die beiden Dati in der Session speicherst und die inputs wieder aus der Session befüllst, funktioniert es auch bei wiederkehr nach einem Wechsel zu anderen Seiten.

**Comment by oioix** [comment]
Ups, ich war zu vorschnell. Ist noch ein Bug drin.

Sofern interesse besteht, mach ich es aber noch gerne fertig.

**Comment by oioix** [comment]
War nur ein Flüchtigkeitsfehler.

hier das Update nun auch mit api + media:

`
// --------------------------------------------------------------------------------------------------------------------------
// -- START DOCUMENT READY --------------------------------------------------------------------------------------------------
$(document).ready(function() {

	//alert('Swissknife script for Redaxo backend is available')
	
	
	// -------------------------------------------------------------------------------------
	// rex addon statistics -> attach filtered date to addon tab links
	// -------------------------------------------------------------------------------------
	var header = $('.rex-page-nav').find('ul').find('a');
	
	var linkStatsElem	= '';
	var linkStatsVal	= '';
	
	var linkPagesElem	= '';
	var linkPagesVal	= '';
	
	var linkRefererElem	= '';
	var linkRefererVal	= '';
	
	var linkApiElem		= '';
	var linkApiVal		= '';
	
	var linkMediaElem	= '';
	var linkMediaVal	= '';
	
	header.each(function(){
		var thisLink = $(this).attr('href');
		
		var inStrStats	= thisLink.includes('statistics/stats');
		var inStrPages	= thisLink.includes('statistics/pages');
		var inStrReferer = thisLink.includes('statistics/referer');
		var inStrApi 	= thisLink.includes('statistics/api');
		var inStrMedia	= thisLink.includes('statistics/media');
		
		if(inStrStats) {
			linkStatsElem = $(this);
			linkStatsVal = thisLink;
		}
		if(inStrPages) {
			linkPagesElem = $(this);
			linkPagesVal = thisLink;
		}
		if(inStrReferer) {
			linkRefererElem = $(this);
			linkRefererVal = thisLink;
		}
		if(inStrApi) {
			linkApiElem = $(this);
			linkApiVal = thisLink;
		}
		if(inStrMedia) {
			linkMediaElem = $(this);
			linkMediaVal = thisLink;
		}
	});
	
	var filterStartDateElem = $('#statistics_datefilter_start');
	var filterEndDateElem	= $('#statistics_datefilter_end');
	
	// get current values
	function getStatFilterDates(){
		var filterStartDate = filterStartDateElem.val();
		var filterEndDate	= filterEndDateElem.val();
	
		if(filterStartDate && filterEndDate) {
			
			var attachDateVal = '&date_start='+filterStartDate+'&date_end='+filterEndDate;
			
			linkStatsElem.attr('href',linkStatsVal+attachDateVal);
			linkPagesElem.attr('href',linkPagesVal+attachDateVal);
			linkRefererElem.attr('href',linkRefererVal+attachDateVal);
			linkApiElem.attr('href',linkApiVal+attachDateVal);
			linkMediaElem.attr('href',linkMediaVal+attachDateVal);
		}
	}
	
	// run on pageload
	getStatFilterDates();
	
	// run on change
	filterStartDateElem.change(function(){
		getStatFilterDates();
	});
	filterEndDateElem.change(function(){
		getStatFilterDates();
	});
});
`

**Comment by alxndr-w** [comment]
@oioix kannst du dazu einen PR erstellen?

**Comment by oioix** [comment]
> @oioix kannst du dazu einen PR erstellen?

Gerade gemacht. In Datei: "statistics.js" eingefügt.

Hab leider im comment versehendlich das falsche Issue "#92" statt "#88" angegeben. Also bitte den comment einfach ignorieren.
JS-solution (see  "Zeitraum filtern" auch für Diagramme https://github.com/AndiLeni/statistics/issues/92)

**Comment by alxndr-w** [comment]
@oioix hier ist kein PR.

**Comment by oioix** [comment]
Ich weiß nicht, was ich da jedes mal falsch mache. Jetzt aber...  

![Screenshot 2024-11-24 | 15 03 14](https://github.com/user-attachments/assets/0a207ad2-65f7-4588-a07b-fd6e41a66da6)

**Comment by alxndr-w** [comment]
Evtl. hast du den PR in deinen lokalen Fork gepackt statt hier. Das kann man beim PR-Erstellen auswählen. Danke schonmal!

**Comment by oioix** [comment]
Ja, vermutlich. Ist mir das letzte mal auch schon passiert. Aber jetzt passt es ja.

**Comment by skerbis** [comment]
Danke für den Hinweis. Das Verhalten ist in PR #136 adressiert: Der gewählte Datumsbereich wird jetzt serverseitig in der Backend-Session gespeichert und beim Tabwechsel wiederverwendet. Dadurch bleibt der Zeitraum über die Statistik-Unterseiten hinweg erhalten.
